tests/test_runtime/test_preciseBN_hook.py

```diff
@@ -0,0 +1,224 @@
+    runner.run(loaders, [('train', 1)], 1)
+
+    # test model w/ gn layer
+    loader = DataLoader(test_bigger_dataset, batch_size=2)
+    loaders = [loader]
+    precise_bn_hook = PreciseBNHook(num_items=4)
+    assert precise_bn_hook.num_items == 4
+    assert precise_bn_hook.interval == 1
+    model = GNExampleModel()
+    runner = EpochBasedRunner(
+        model=model, batch_processor=None, optimizer=optimizer, logger=logger)
+    runner.register_hook(precise_bn_hook)
+    runner.run(loaders, [('train', 1)], 1)
+
+    # test model without bn layer
+    loader = DataLoader(test_bigger_dataset, batch_size=2)
+    loaders = [loader]
+    precise_bn_hook = PreciseBNHook(num_items=4)
+    assert precise_bn_hook.num_items == 4
+    assert precise_bn_hook.interval == 1
+    model = NoBNExampleModel()
+    runner = EpochBasedRunner(
+        model=model, batch_processor=None, optimizer=optimizer, logger=logger)
+    runner.register_hook(precise_bn_hook)
+    runner.run(loaders, [('train', 1)], 1)
+
+    # test how precise it is
+    loader = DataLoader(test_bigger_dataset, batch_size=2)
+    loaders = [loader]
+    precise_bn_hook = PreciseBNHook(num_items=12)
+    assert precise_bn_hook.num_items == 12
+    assert precise_bn_hook.interval == 1
+    model = SingleBNModel()
+    runner = EpochBasedRunner(
+        model=model, batch_processor=None, optimizer=optimizer, logger=logger)
+    runner.register_hook(precise_bn_hook)
+    runner.run(loaders, [('train', 1)], 1)
+    imgs_list = list()
+    for loader in loaders:
+        for i, data in enumerate(loader):
+            imgs_list.append(np.array(data['imgs']))
+    mean = np.mean([np.mean(batch) for batch in imgs_list])
+    # bassel correction used in Pytorch, therefore ddof=1
+    var = np.mean([np.var(batch, ddof=1) for batch in imgs_list])
+    assert np.equal(mean, np.array(
+        model.bn.running_mean)), (mean, np.array(model.bn.running_mean))
+    assert np.equal(var, np.array(
+        model.bn.running_var)), (var, np.array(model.bn.running_var))
+
+    @pytest.mark.skipif(
+        not torch.cuda.is_available(), reason='requires CUDA support')
+    def test_ddp_model_precise_bn():
+        # test DDP model
+        test_bigger_dataset = BiggerDataset()
+        loader = DataLoader(test_bigger_dataset, batch_size=2)
+        loaders = [loader]
+        precise_bn_hook = PreciseBNHook(num_items=5)
+        assert precise_bn_hook.num_items == 5
+        assert precise_bn_hook.interval == 1
+        model = ExampleModel()
+        model = MMDistributedDataParallel(
+            model.cuda(),
+            device_ids=[torch.cuda.current_device()],
+            broadcast_buffers=False,
+            find_unused_parameters=True)
+        runner = EpochBasedRunner(
+            model=model,
+            batch_processor=None,
+            optimizer=optimizer,
+            logger=logger)
+        runner.register_hook(precise_bn_hook)
+        runner.run(loaders, [('train', 1)], 1)
```

Comment:
Please set `max_epochs` when instantiating the runner instead of passing it in `run`.

Reply:
done.